changelogs/fragments/2383-influxdb_retention_policy-add-state-option.yml

```
@@ -0,0 +1,8 @@
+minor_changes:
+  - influxdb_retention_policy - add ``state`` parameter with allowed values
+    ``present`` and ``absent`` to support deletion of existing retention policies
+    (https://github.com/ansible-collections/community.general/issues/2383).
+  - influxdb_retention_policy - simplify duration logic parsing as suggested in
+    https://github.com/ansible-collections/community.general/pull/2284#discussion_r618210902
+  - influxdb_retention_policy - fix bug where ``INF`` duration values failed parsing,
+    introduced in https://github.com/ansible-collections/community.general/pull/2284
```

```suggestion
  - influxdb_retention_policy - simplify duration logic parsing
    (https://github.com/ansible-collections/community.general/pull/2385).
bugfixes:
  - influxdb_retention_policy - fix bug where ``INF`` duration values failed parsing
    (https://github.com/ansible-collections/community.general/pull/2385).
```
I would suggest to split out the bugfix into a separate PR so that this can be backported and also be used in a 3.0.1 release. If there's a feature in the PR, it can only be used in 3.1.0, but not be backported.

---

Done, a new PR has been issued, #2396